common-test/src/main/java/io/customer/commontest/config/TestConfigExtensions.kt

```diff
@@ -0,0 +1,58 @@
+package io.customer.commontest.config
+
+import io.customer.sdk.core.di.AndroidSDKComponent
+import io.customer.sdk.core.di.SDKComponent
+
+/**
+ * Typealias for test configuration DSL for improved readability and ease of use.
+ */
+typealias ConfigDSL<Config> = Config.() -> Unit
+
+/**
+ * Combines two [ConfigDSL]s into a single [ConfigDSL] that will execute both of them in sequence.
+ * This is useful for combining multiple configurations into a single configuration
+ * and mimics the behavior of overriding method in a class.
+ */
+operator fun <Config> ConfigDSL<Config>.plus(other: ConfigDSL<Config>): ConfigDSL<Config> = {
+    this@plus()
+    other()
+}
+
+/**
+ * Retrieves the first argument of type [Arg] from test configuration arguments.
+ * If no argument of type [Arg] is found, returns null.
+ * If more than one argument of type [Arg] is found, throws an [IllegalArgumentException].
+ */
+inline fun <reified Arg : TestArgument> TestConfig.argumentOrNull(): Arg? {
+    val args = arguments.filterIsInstance<Arg>()
+    return when {
+        args.isEmpty() -> return null
+        args.size == 1 -> args[0]
+        else -> throw IllegalArgumentException("Expected exactly one argument of type ${Arg::class.java.simpleName}, but found ${args.size}")
+    }
+}
```

Comment:
```suggestion
inline fun <reified Arg : TestArgument> TestConfig.argumentOrNull(): Arg? =
    arguments.filterIsInstance<Arg>().singleOrNull()
```

Reply:
Aah, I kept looking for similar method and literally forgot about this one 🤦🏻 Thanks for sharing